src/openassetio-core-c/include/openassetio/c/SimpleMap.h

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2013-2022 The Foundry Visionmongers Ltd
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+
+#include <openassetio/c/export.h>
+
+#include "./StringView.h"
+#include "./namespace.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+/**
+ * @addtogroup CAPI C API
+ * @{
+ */
+
+/**
+ * Opaque handle type representing a @fqref{SimpleMap} "SimpleMap"
+ * instance.
+ *
+ * The associated @fqcref{SimpleMap_s} "SimpleMap suite" functions
+ * operate on this handle type.
+ */
+// NOLINTNEXTLINE(modernize-use-using)
+typedef struct OPENASSETIO_NS(SimpleMap_t) * OPENASSETIO_NS(SimpleMap_h);
+
+/**
+ * Function pointer suite for a @fqref{SimpleMap} "SimpleMap" C API.
+ *
+ * Instances of this suite are provided by the `SimpleMap_suite`
+ * function and operate on a @fqcref{SimpleMap_h} "SimpleMap handle".
+ */
+// NOLINTNEXTLINE(modernize-use-using)
+typedef struct {
```

Comment:
Using the 'suite' of function pointers seems a bit overkill. Is this strictly necessary, or could we use simple linkage?

The times when using a fptr-suite is most useful is for runtime loadable plugin systems.
1. The factory method can hide the discovery mechanism and loading of plugins
2. The suite object provides a convenient way for plugins to 'self describe' their entry-points after loading (e.g. through a factory function of their own)
3. A single plugin can implement several different public APIs by providing different 'levels' of suites with different (but usually overlapping) behaviours.

Do we need any of that?

My preference would be just to have plain extern-"C" functions:
- It would be up to the user at link time to decide to use static or dynamic linking, and both would work
- In the static linking case, unused code would be strippable.
- Doesn't need any boilerplate-code on the library side or the user's side for accessing or setting up suite objects.

Reply:
Good points well made and we do tend to agree so will refactor this.  As you mention, function suites are a valuable pattern for plugins, but not so much for the host (Katana notwithstanding).

For reference, we noted some advantages of function pointer suites in [C_API.md](https://github.com/TheFoundryVisionmongers/OpenAssetIO/blob/feature/181-coreCppBuild/contributing/C_API.md), which I'll reproduce below:

Such suites have a few advantages
* Aesthetically, a struct of function pointers mirrors a class with
  methods, and if the suite is providing a C API to a C++ class then
  mentally mapping one to the other is straightforward.
* Only the suite type needs namespacing, the "methods" are simply
  struct members, which reduces code verbosity.
* Since a suite is an instance of a struct, it must be provided by
  a function call, allowing the provider a great deal of flexibility
  in its construction, compared to link-time defined symbols.
* The suite is an object that can be passed around, including as input
  to a wrapper C++ class (see below).
* In C++ the suite's functions can be implemented as a series of 
  (non-capturing) lambdas, keeping related code grouped together.